frontend/src/components/CookRecipe.js

```diff
@@ -92,12 +111,74 @@ class CookRecipe extends Component {
             </div>
           </Tab>
           <Tab eventKey="tutorial" title="Tutorial">
-            {/* to do */}
-            <div className="tab-content"></div>
+            <div className="tab-content">
+              <Carousel
+                interval={null} // to disable auto play of the carousel
+                onSelect={this.setSelectedStep}
+                defaultActiveIndex={this.getSelectedStep}
+                nextIcon={
+                  <img
+                    src={navigateNext}
+                    alt="next step"
+                    className="carousel-control"
+                  />
+                }
+                prevIcon={
+                  <img
+                    src={navigatePrevious}
+                    alt="previous step"
+                    className="carousel-control"
+                  />
+                }
+              >
+                {recipe.instructions.map((step, i) => (
+                  <Carousel.Item key={i} className="carousel-step">
+                    {step}
+                    <Button variant="" onClick={this.readStep}>
+                      <img src={speaker} alt="read the step" />
+                    </Button>
+                  </Carousel.Item>
+                ))}
+              </Carousel>
+              {this.displayFinishedIfLastStep()}
+            </div>
           </Tab>
         </Tabs>
       </div>
     );
   }
+
+  setSelectedStep = (selectedIndex, e) => {
+    localStorage.setItem("tutorial-step", selectedIndex);
+    this.checkIsLastStep();
```

Comment:
nit: rename to maybeDoLastStepProcessing() since this is doing more than checking if this is the last step.